PyTAPS/transports.py

```diff
@@ -0,0 +1,356 @@
+        """except:
+            print_time("SendError occured.", color)
+            if self.connection.send_error:
+                self.loop.create_task(self.connection.send_error(self.message_count))
+            return"""
+        print_time("Data written successfully.", color)
+        if self.connection.sent:
+            self.loop.create_task(self.connection.sent(self.message_count))
+        return
+
+    async def close(self):
+        print_time("Closing connection.", color)
+        self.transport.close()
+        self.connection.state = ConnectionState.CLOSED
+        if self.connection.closed:
+            self.loop.create_task(self.connection.closed())
+
+    async def read(self, min_incomplete_length, max_length):
+        print_time("Reading message", color)
+        if self.recv_buffer is None:
+            await self.await_data()
+        print_time("Received full message", color)
+        if len(self.recv_buffer) == 1:
+            data = self.recv_buffer[0]
+            self.recv_buffer = None
+        else:
+            data = self.recv_buffer.pop(0)
+        if self.connection.received:
+            self.loop.create_task(self.connection.received(data.decode(),
+                                                           "Context",
+                                                           self.connection))
+
+    # Asyncio Callbacks
+
+    """ ASYNCIO function that gets called when a new
+        connection has been made, similar to TAPS ready callback.
+    """
+    def connection_made(self, transport):
+        if self.connection.state == ConnectionState.ESTABLISHED:
+            transport.close()
+            return
+        # Check if its an incoming or outgoing connection
+        if self.connection.active is False:
+            self.transport = transport
+            new_remote_endpoint = RemoteEndpoint()
+            print_time("Received new connection.", color)
+            # Get information about the newly connected endpoint
+            new_remote_endpoint.with_address(
+                                transport.get_extra_info("peername")[0])
+            new_remote_endpoint.with_port(
+                                transport.get_extra_info("peername")[1])
+            self.remote_endpoint = new_remote_endpoint
+            self.connection.state = ConnectionState.ESTABLISHED
+            if self.connection.connection_received:
+                self.loop.create_task(self.connection_received(self))
+            return
+
+        elif self.connection.active:
+            """
+            for t in self.connection.transports:
+                if t != self:
+                    self.connection.transports.remove(t)
+
+            for t in self.connection.pending.keys():
+                if self.connection.pending[t] != self:
+                    print(self.connection.pending[t])
+                    print(t)
+                    t.cancel()"""
```

Comment:
Can we remove this dead code?

Reply:
These parts are alternatives on how to deal with the other transports after a connection has already been established. This part here would kill the asyncio task instead of waiting for it to complete and then closing the connection like we are doing now. I left it in because we might want to change to this way in the future.

Reply:
I see, then I'm fine with keeping it in for now.
Please add a comment briefly explaining what this does.